pkg/controller/gcppubsub/receive_adapter/receive_adapter.go

```diff
@@ -0,0 +1,105 @@
+package main
+
+import (
+	"flag"
+	"fmt"
+	"io/ioutil"
+	"log"
+	"net/http"
+	"os"
+
+	"go.uber.org/zap"
+
+	// Imports the Google Cloud Pub/Sub client package.
+	"cloud.google.com/go/pubsub"
+	"github.com/knative/eventing/pkg/event"
+	"golang.org/x/net/context"
+)
+
+const (
+	// Environment variable containing project id
+	envProject = "GCPPUBSUB_PROJECT"
+
+	// Sink for messages.
+	envSinkURI = "SINK_URI"
+
+	envTopic = "GCPPUBSUB_TOPIC"
+
+	// Name of the subscription to use
+	envSubscription = "GCPPUBSUB_SUBSCRIPTION_ID"
+)
+
+func main() {
+	flag.Parse()
+
+	logger, err := zap.NewProduction()
+	if err != nil {
+		log.Fatalf("Unable to create logger: %v", err)
+	}
+
+	projectID := getRequiredEnv(envProject)
+	topicID := getRequiredEnv(envTopic)
+	sinkURI := getRequiredEnv(envSinkURI)
+	subscriptionID := getRequiredEnv(envSubscription)
+
+	logger.Info("Starting.", zap.String("projectID", projectID), zap.String("topicID", topicID), zap.String("subscriptionID", subscriptionID), zap.String("sinkURI", sinkURI))
+
+	ctx := context.Background()
+	source := fmt.Sprintf("//pubsub.googleapis.com/%s/topics/%s", projectID, topicID)
+
+	// Creates a client.
+	// TODO: Support additional ways of specifying the credentials for creating.
+	client, err := pubsub.NewClient(ctx, projectID)
+	if err != nil {
+		logger.Fatal("Failed to create client: %v", zap.Error(err))
+	}
+
+	sub := client.Subscription(subscriptionID)
+
+	err = sub.Receive(ctx, func(ctx context.Context, m *pubsub.Message) {
```

Comment:
`sub.Receive` "blocks until ctx is done, or the service returns a non-retryable error."

The normal usage of `sub.Receive` is to pass it a cancelable context. Then cancel from the outside when appropriate. But because all this program is doing is receiving messages (and we don't have any shutdown logic), there is no reason to call cancel. 

We probably should add graceful shutdown logic, but I would rather do that in the future than in this PR. At worst what will happen is a message will be processed multiple times (i.e. it is sent downstream, but not ACKed, and the RA is shutdown, the new RA sends the same message downstream again).

Reply:
I created #54 to track this enhancement.